src/mlpack/methods/ann/layer/add.hpp

```diff
@@ -97,6 +97,11 @@ class Add
   //! Modify the gradient.
   OutputDataType& Gradient() { return gradient; }
 
+  //! Get the layer name.
+  const std::string& Name() const { return name; }
+  //! Modify the layer name.
+  std::string& Name() { return name; }
```

Comment:
I did it so that names can be customized like `CONV4` referring to convolution layer in 4th layer or something like that. I saw pattern like this in model summaries of keras models like vgg and inception.
I can remove it if it's not that necessary.

Reply:
For that we could use the index, but I don't think modifying the name of the layer is the best approach.